session.go

```diff
@@ -347,10 +351,23 @@ func (s *session) persist(seqNum int, msgBytes []byte) error {
 }
 
 func (s *session) sendQueued() {
-	for _, msgBytes := range s.toSend {
-		s.sendBytes(msgBytes)
+	var (
+		sent         bool
+		indexBlocked int
+	)
+
+	for i, msgBytes := range s.toSend {
+		sent = s.sendBytes(msgBytes)
+		if !sent {
+			indexBlocked = i
+			break
+		}
+	}
+	if !sent {
+		s.toSend = s.toSend[indexBlocked:]
+		s.notifyMessageOut()
+		return
```

Comment:
minor: could be simplified now:

```suggestion
	for i, msgBytes := range s.toSend {
		if !s.sendBytes(msgBytes) {
			s.toSend = s.toSend[i:]
			s.notifyMessageOut()
			return
		}
```

Reply:
yes yes, I avoided this because we override `s.toSend` while ranging over it. 
Seemed odd to me. @Neal is it fine if we keep it as it is?

Reply:
it is not an issue to overwrite, especially in this case since we return right after. I'm fine either way.

Reply:
Tested and it works fine.

Addressed: https://github.com/alpacahq/quickfix/pull/18/commits/9693665cce97725f993ad72c53205846d1022653